api/web3server.go

```diff
@@ -617,6 +623,71 @@ func (svr *Web3Server) getTransactionByHash(in interface{}) (interface{}, error)
 	return svr.getTransactionCreateFromActionInfo(actionInfos)
 }
 
+func (svr *Web3Server) debugTransactionByHash(in interface{}) (interface{}, error) {
+	h, err := getStringFromArray(in, 0)
+	if err != nil {
+		return nil, err
+	}
+	actInfo, err := svr.coreService.Action(util.Remove0xPrefix(h), false)
+	if err != nil {
+		st, ok := status.FromError(err)
+		if ok && st.Code() == codes.Unavailable {
+			return nil, nil
+		}
+		return nil, err
+	}
+	nselp := &action.SealedEnvelope{}
+	var act *action.Execution
+
+	switch actInfo.Action.Core.Action.(type) {
```

Comment:
1. we only care a bout Execution here, so can use if instead of switch
2. no need to call `nselp.LoadProto`, can directly use protobuf data field
```
if exec, ok := actInfo.Action.Core.Action.(*iotextypes.ActionCore_Execution); !ok {
   return nil, errors.Errorf("the type of action %s is not supported", actInfo.ActHash)
}
// use exec.Amount, exec.Contract, exec.Data
```